src/api.ts

```diff
@@ -0,0 +1,185 @@
+const base = 'http://localhost:3000';
+
+const garage = `${base}/garage`;
+const engine = `${base}/engine`;
+const winners = `${base}/winners`;
+
+export const getCars = async (
+    page: number,
+    limit = 7
+): Promise<{
+    items: { id: number; color: string; name: string }[];
+    count: string;
+}> => {
+    const response = await fetch(`${garage}?_page=${page}&_limit=${limit}`);
+    return {
+        items: await response.json(),
+        count: <string>response.headers.get('X-Total-Count'),
+    };
+};
+
+export const getCar = async (id: number): Promise<{ id: number; color: string; name: string }> =>
+    (await fetch(`${garage}/${id}`)).json();
+
+export const createCar = async (data: { color: string; name: string }): Promise<JSON> =>
```

Comment:
The returned Promise type is not correct. We need to specify the data type that will be returned by the API. JSON is an object that provides functions to work with json. It won't be an actual data